pkg/local_object_storage/blobstor/get.go

```diff
@@ -12,47 +12,35 @@ import (
 // Get reads the object from b.
 // If the descriptor is present, only one sub-storage is tried,
 // Otherwise, each sub-storage is tried in order.
-func (b *BlobStor) Get(addr oid.Address, storageID []byte) (*objectSDK.Object, error) {
+func (b *BlobStor) Get(addr oid.Address) (*objectSDK.Object, error) {
 	b.modeMtx.RLock()
 	defer b.modeMtx.RUnlock()
 
-	if storageID == nil {
-		for i := range b.storage {
-			obj, err := b.storage[i].Storage.Get(addr)
-			if err == nil || !errors.As(err, new(apistatus.ObjectNotFound)) {
-				return obj, err
-			}
+	for i := range b.storage {
```

Comment:
why do you want to merge it now? it does some functional changes. once peapod is dropped, this PR would do more correct changes, and there is no magic `for` over every storage we have. this PR moves master to some strange state, i do not agree with it

Reply:
What are the functional changes here? Yes, substorages are iterated over, but it's the same if we don't pass any storage ID. And practically we have one substorage per shard everywhere except for three machines (and those to be converted soon).

This cuts one of the problematic things from the codebase, then we can move on to the next one, not trying to solve everything in a single PR.